src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -0,0 +1,155 @@
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
+
+        public override Symbol? AssociatedSymbol => null;
+
+        public override Symbol ContainingSymbol => ContainingType;
+
+        public override ImmutableArray<Location> Locations => ContainingType.Locations;
+
+        public override Accessibility DeclaredAccessibility => Accessibility.Public;
+
+        public override bool IsStatic => false;
+
+        public override bool IsVirtual => false;
+
+        public override bool IsOverride => false;
+
+        public override bool IsAbstract => false;
+
+        public override bool IsSealed => false;
+
+        public override bool IsExtern => false;
+
+        internal override bool HasSpecialName => false;
+
+        internal override LexicalSortKey GetLexicalSortKey() => LexicalSortKey.SynthesizedRecordEquals;
+
+        internal override MethodImplAttributes ImplementationAttributes => MethodImplAttributes.Managed;
+
+        internal override bool HasDeclarativeSecurity => false;
+
+        internal override MarshalPseudoCustomAttributeData? ReturnValueMarshallingInformation => null;
+
+        internal override bool RequiresSecurityObject => false;
+
+        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+
+        internal override bool GenerateDebugInfo => false;
+
+        public override DllImportData? GetDllImportData() => null;
+
+        internal override ImmutableArray<string> GetAppliedConditionalSymbols()
+            => ImmutableArray<string>.Empty;
+
+        internal override IEnumerable<SecurityAttribute> GetSecurityInformation()
+            => Array.Empty<SecurityAttribute>();
+
+        internal override bool IsMetadataNewSlot(bool ignoreInterfaceImplementationChanges = false) => false;
+
+        internal override bool IsMetadataVirtual(bool ignoreInterfaceImplementationChanges = false) => false;
+
+        internal override bool SynthesizesLoweredBoundBody => true;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var F = new SyntheticBoundNodeFactory(this, ContainingType.GetNonNullSyntaxNode(), compilationState, diagnostics);
+
+            // Compare all of the record properties in this class with the argument properties
+            // Body:
+            // {
+            //      return other != null && `comparisons`;
+            // }
+
+            var other = F.Parameter(Parameters[0]);
+            BoundExpression? retExpr = null;
+            if (!ContainingType.IsStructType())
+            {
+                retExpr = F.ObjectNotEqual(other, F.Null(F.SpecialType(SpecialType.System_Object)));
+            }
+
+            var recordProperties = ArrayBuilder<FieldSymbol>.GetInstance();
+            foreach (var member in ContainingType.GetMembersUnordered())
```

Comment:
>GetMembersUnordered [](start = 50, length = 19)

Generating code based on the unordered ordering seems fragile.  I suggest they should be in the order of the original declarations.